### PR TITLE
JI:768392 P2022-1542 Summary of the finished presentation

### PR DESCRIPTION
**Pull request by Adys23**

### CODE REVIEW

File: libs/shared/src/components/form/index.tsx [hunk 1]
```diff
@@ -7,14 +7,15 @@ import { useTranslation } from "react-i18next";
...
 
+import { FormFieldProps } from "../../types";
 import { BaseButton, ButtonType } from "../base-button";
-import { FormField, FormFieldProps } from "./form-field";
+import { FormField } from "./form-field";
 import * as Styled from "./styled";
 
 interface FormButton {
   condition: boolean;
   text: string;
```
Comment (aniasgit) [inline libs/shared/src/components/form/index.tsx:17]: The text can also be optional, we have some default values for it.

File: libs/ui/src/stories/form/form.stories.tsx [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
-    fields={fields}
-    onSubmit={onSubmit}
-    onError={onError}
-    showSubmitButton={showSubmitButton}
-  />
+export const FormStory: React.FC<FormProps> = ({ validationSchema, fields, onSubmit, onError, customButtons }) => (
+  <Form validationSchema={validationSchema} fields={fields} onSubmit={onSubmit} onError={onError} customButtons={customButtons} />
 );
 
+const customButtons: FormProps["customButtons"] = { submit: { condition: true, text: t("submit") } };
```
Comment (aniasgit) [inline libs/ui/src/stories/form/form.stories.tsx:11]: You should change text in interface for optional, and t("submit") won't be necessary.

File: libs/features/feedback/src/components/external-user-presentation-summary/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+export interface ExtPresentationSummaryProps {
+  questions: Question[];
+}
+
+interface PresentationResultsData {
+  labels: string[];
+  values: number[];
+}
+
+export const ExternalUserPresentationSummary: React.FC<ExtPresentationSummaryProps> = ({ questions }) => {
```
Comment (Reden4005) [inline libs/features/feedback/src/components/external-user-presentation-summary/index.tsx:17]: ExternalUserPresentationSummaryProps maybe? Or just PresentationSummary and PresentationSummaryProps?

File: libs/features/feedback/src/components/external-user-presentation-summary/index.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    return (
+      <Styled.DiagramContainer key={id}>
+        <Diagram title={content} type={DiagramType.Bar} labels={labels} values={values} />
+      </Styled.DiagramContainer>
+    );
+  });
+
+  return (
+    <Styled.PresentationSummaryContainer>
+      <Typography variant='h6' className='summaryTitle'>
```
Comment (Reden4005) [inline libs/features/feedback/src/components/external-user-presentation-summary/index.tsx:41]: We should use styled components here.

File: libs/features/feedback/src/components/external-user-presentation-summary/styled.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import { styled } from "@mui/material/styles";
+
+export const PresentationSummaryContainer = styled(Box)(({ theme }) => ({
+  display: "flex",
+  flexDirection: "column",
+  width: "100%",
+  alignItems: "center",
+  margin: theme.spacing(1),
+
+  "& .summaryTitle": {
```
Comment (Reden4005) [inline libs/features/feedback/src/components/external-user-presentation-summary/styled.ts:11]: https://github.com/intive/patronage22-web-wroclaw/pull/75/files#r841954758

File: libs/ui/src/stories/external-user-presentation-summary/external-user-presentation-summary.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+      { answer: 5, content: "Cucumber", count: 17 }
+    ]
+  },
+  {
+    id: "3",
+    content: "Best SW movie:",
+    answers: [
+      { answer: 1, content: "Episode I", count: 10 },
+      { answer: 2, content: "Episode II", count: 13 },
+      { answer: 3, content: "Episode III", count: 25 },
```
Comment (krysia1) [inline libs/ui/src/stories/external-user-presentation-summary/external-user-presentation-summary.stories.tsx:37]: Interesting results 😝

Comment (Adys23) [reply libs/ui/src/stories/external-user-presentation-summary/external-user-presentation-summary.stories.tsx:37]: image

File: libs/features/feedback/src/components/external-user-presentation-summary/index.tsx (hunk 3 above)
Comment (Adys23) [reply libs/features/feedback/src/components/external-user-presentation-summary/index.tsx:17]: I was thinking about PresentationSummaryProps, but not sure if we will use same component for external and internal use.

Comment (kacperChlon) [reply libs/features/feedback/src/components/external-user-presentation-summary/index.tsx:17]: PresentationLiveSummary <- for external user summary
PresentationSummary <- for admin which will be able to see results from his dashboard

File: libs/features/feedback/src/components/external-user-presentation-summary/index.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+        acc.values.push(item.count);
+
+        return acc;
+      },
+      { labels: [], values: [] }
+    );
+
+    return (
+      <Styled.DiagramContainer key={id}>
+        <Diagram title={content} type={DiagramType.Bar} labels={labels} values={values} />
```
Comment (kacperChlon) [inline libs/features/feedback/src/components/external-user-presentation-summary/index.tsx:34]: const diagramValues = answers.reduce<P...

<Diagram ..., {...diagramValues} />

File: libs/features/feedback/src/components/external-user-presentation-summary/index.tsx (hunk 4 above)
Comment (kacperChlon) [reply libs/features/feedback/src/components/external-user-presentation-summary/index.tsx:41]: Correct :)

File: libs/shared/src/components/form/index.tsx (hunk 1 above)
Comment (Adys23) [reply libs/shared/src/components/form/index.tsx]: Done

File: libs/ui/src/stories/form/form.stories.tsx (hunk 2 above)
Comment (Adys23) [reply libs/ui/src/stories/form/form.stories.tsx]: Done

File: libs/features/feedback/src/components/external-user-presentation-summary/index.tsx (hunk 3 above)
Comment (Adys23) [reply libs/features/feedback/src/components/external-user-presentation-summary/index.tsx]: Sure

File: libs/features/feedback/src/components/external-user-presentation-summary/index.tsx (hunk 4 above)
Comment (Adys23) [reply libs/features/feedback/src/components/external-user-presentation-summary/index.tsx]: :D done

File: libs/features/feedback/src/components/external-user-presentation-summary/styled.ts (hunk 5 above)
Comment (Adys23) [reply libs/features/feedback/src/components/external-user-presentation-summary/styled.ts]: done

File: libs/features/feedback/src/components/external-user-presentation-summary/index.tsx (hunk 7 above)
Comment (Adys23) [reply libs/features/feedback/src/components/external-user-presentation-summary/index.tsx]: done

File: libs/features/feedback/src/components/presentation-live-summary/index.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  questions: Question[];
+}
+
+interface PresentationResultsData {
+  labels: string[];
+  values: number[];
+}
+
+export const PresentationLiveSummary: React.FC<PresentationLiveSummaryProps> = ({ questions }) => {
+  const { t } = useTranslation([TranslationNamespace.Feedback]);
```
Comment (aniasgit) [inline libs/features/feedback/src/components/presentation-live-summary/index.tsx:17]: If there is only one nemespace [] are no needed

**Review by kacperChlon — APPROVED** [review]

File: libs/features/feedback/src/components/presentation-live-summary/index.tsx (hunk 8 above)
Comment (Adys23) [reply libs/features/feedback/src/components/presentation-live-summary/index.tsx]: Done

**Pull request merged**